File: C#/Solutions/Problem005/Pair.cs
namespace DailyCodingProblems.Solutions.Problem005
{
    /// <summary>
    /// Creates a pair of <see cref="T"/>
    /// </summary>
    /// <typeparam name="T"></typeparam>
    public class Pair<T>
    {
        private readonly T _a;
        private readonly T _b;

        private Pair(T a, T b)
        {
            _a = a;
            _b = b;
        }

        public static Pair<T> Cons(T a, T b)
        {
            return new Pair<T>(a, b);
        }

        public static T Car(Pair<T> pair)
        {
            return pair._a;
        }

        public static T Cdr(Pair<T> pair)
        {
            return pair._b;
        }
    }
}

File: Java/Solutions/Problem005/Pair.java
package Problem005;

/**
 * Creates a pair of T
 */
public class Pair<T> {
    private T _a;
    private T _b;

    private Pair(T a, T b) {
        _a = a;
        _b = b;
    }

    public static <T> Pair<T> cons(T a, T b) {
        return new Pair<T>(a, b);
    }

    public static <T> T car(Pair<T> pair) {
        return pair._a;
    }

    public static <T> T cdr(Pair<T> pair) {
        return pair._b;
    }
}
